### ISSUE TITLE
npm install does not update the modification time (mtime) of node_modules folder

### ISSUE THREAD

**Issue by wil93** [npm/npm#10681]
I have a `Makefile` like this:

    node_modules: package.json
    	npm install

The problem is, `npm install` will always be executed (because `package.json` is always newer than `node_modules`). This is probably caused by `npm` not updating the modification time of `node_modules`.

To work around it, I did this:

    node_modules: package.json
    	npm install
    	touch node_modules

**Comment by zkat** [comment]
This seems like it would be fairly easy to add, and a fantastic contribution. We like it! Please submit a patch if you're still interested in this!

In the meantime, I'm closing this particular issue.

**Comment by wil93** [comment]
I'm interested in this, but I wouldn't know where to start to patch it :confused:

Note, however, that if the issue is closed then it will take some other user to encounter this problem again before it gets implemented (unless there's someone who loves browsing closed issues tagged with "patch-welcome" :smile:). In my head a closed issue means fixed/wontfix, so I would have left it open (and tagged, so it's easy to find for someone who's looking to start contributing). But whatever :smile:

However, I'll see if it's easy to do and try to do it :sweat_smile:

**Comment by othiym23** [comment]
> In my head a closed issue means fixed/wontfix, so I would have left it open (and tagged, so it's easy to find for someone who's looking to start contributing).

That's true. On the other hand, there are around 400 open feature requests on npm, and there's no way the project team is going to have the time in the conceivable future to get to them (the last time I asked @iarna how long she thought it would take us to implement everything that had made it into the project roadmap, she said, "ten years", which sounds remarkably like "forever" to me when I'm wearing my project manager's hat). Leaving feature requests open is misleading, because it leads people to believe that our team plans to implement them, when in fact we're simply not going to have the time.

Basically, an issue tracker with 1800 open issues on it is not very useful for anyone, and we're doing what we think we must to get things under control. Thanks for your understanding, and do keep in mind that we're happy to answer questions and work with people to get patches into good enough shape to land.

**Comment by wil93** [comment]
Hi :smile: came back to say that I just noticed how the Go project is managing the high number of issues in [their tracker](https://github.com/golang/go/issues) by using milestones. As you can see, they currently have just over 2000 issues open, but 1200+ of them are in the "Unplanned" milestone. If you filter by "Go1.6" milestone (which will be released soon) you get just 4 issues, and if you filter by "Go1.7" you get 180.

I think the npm project could use this method as well, since it would only require an initial decision ("what milestone do I put this issue?").

Note that I'm *not* asking to reopen this or any issue, I'm just suggesting to consider using milestones in order to not lose information (e.g. mixing "wontfix/fixed" issues with "unplanned" ones :wink:)

**Comment by othiym23** [comment]
That's an interesting idea that was similar to something the CLI team tried for a while, but it doesn't scale. When we realized we had about 10 years' worth of work sitting in the "unplanned" bucket, we decided we needed to try something else.

Because npm is built on semver, we don't really do release milestones per se. Mostly, we figure out whether the next release is going to be breaking, a feature release, or a patch release when it's time to do the weekly release, and the situation we kept running into was that stuff kept getting bumped from release to release (and also, there was way more of it piling up than we actually thought we could get to in a reasonable timeframe).

None of this process is immutable; we're constantly tweaking how we do things to match what we need to do and what the community is doing. If we see a major change in how people are engaging with our issues, we can address this again. As it stands, if anything the amount of outside contribution has gone up sharply recently, and I think the renewed emphasis on sorting through feature requests is a big part of that.
